lib/kitchen/provisioner/chef_base.rb

```diff
@@ -265,6 +267,74 @@ def reload_ps1_path
           %{[System.Environment]::GetEnvironmentVariable("PATH","Machine")\n\n}
         ].join(" = ")
       end
+
+      # Writes a configuration file to the sandbox directory.
+      # @api private
+      def prepare_config_rb
+        data = default_config_rb.merge(config[config_filename.gsub(".", "_").to_sym])
+
+        info("Preparing #{config_filename}")
+        debug("Creating #{config_filename} from #{data.inspect}")
+
+        File.open(File.join(sandbox_path, config_filename), "wb") do |file|
+          file.write(format_config_file(data))
+        end
+
+        prepare_config_idempotency_check if config[:enforce_idempotency]
+      end
+
+      # Writes a configuration file to the sandbox directory
+      # to check for idempotency of the run.
+      # @api private
+      def prepare_config_idempotency_check
+        handler_filename = "chef-client-fail-if-update-handler.rb"
+        source = File.join(
+          File.dirname(__FILE__), %w[.. .. .. support ], handler_filename
+        )
+        FileUtils.cp(source, File.join(sandbox_path, handler_filename))
+        File.open(File.join(sandbox_path, "client_no_updated_resources.rb"), "wb") do |file|
+          file.write(format_config_file(data))
+          file.write("\n\n")
+          file.write("handler_file = File.join(File.dirname(__FILE__), '#{handler_filename}')\n")
+          file.write "Chef::Config.from_file(handler_file)\n"
+        end
+      end
+
+      # Returns an Array of command line arguments for the chef client.
+      #
+      # @return [Array<String>] an array of command line arguments
+      # @api private
+      def chef_args(_config_filename)
```

Comment:
dont need the `_config_filename` param here anymore since the method can just use the new `config_filename` method.